lib/core/utils/flattened-tree.js

```diff
@@ -28,10 +28,11 @@ var axe = axe || { utils: {} };
  */
 function virtualDOMfromNode(node, shadowId) {
 	const vNodeCache = {};
-	return {
+	const vNode = {
 		shadowId: shadowId,
 		children: [],
 		actualNode: node,
+		_isHidden: null, // will be populated by axe.utils.isHidden
```

Comment:
I see what you are doing here, but to stay with the `getter/setter` pattern, should we introduce `set isHidden(value)` & `get isHidden()` which them maintains `_isHidden` with in `vNodeCache`.

This will avoid what looks like accessing an internal property like `_isHidden` from  `axe.utils.isHidden` & keeps things neat.

Reply:
I like the getter/setter idea, though with the `isVisible` cache we would need 3 different getters/setters, 4 if we include `isHiddenWithCss`. I would suggest we decided what we want to do with the 3 different functions to check hidden state before we go ahead and make a ton of getter/setters we might not need later.

Reply:
Created issue to talk about this - https://github.com/dequelabs/axe-core/issues/1519. Not a blocker for this PR.